annotation/main.scm

```diff
@@ -47,24 +48,23 @@
    find-genes GENE-LIST
    Validate if given gene strings in GENE-LIST exist in the atomspace.
 "
-  (let ((unknown (filter (lambda (gene)
-                           (null? (cog-node 'GeneNode gene)))
-                         gene-list)))
-    (match unknown
-      (() (check-outdate-genes gene-list))
-      (_ 
-        (let* (
-          (res (append-map find-similar-gene unknown))
-          (suggestions (if (> (length res) 5) (map (lambda (u) (cog-name u)) (take res 5)) (map (lambda (u) (cog-name u)) res))
-             )
+  (let* ((records (filter-map (lambda (g)
+                    (if (null? (cog-node 'GeneNode g))
+                        (make-gene g (make-gene-info "" (find-similar-gene g)))
+                        (let* ([curr (find-current-symbol g)])
+                          (if (null? curr)
+                            #f
+                            (make-gene g (make-gene-info (car curr) '()))
```

Comment:
I am getting an error as the (find-current-symbol g) returns an opencog node (if a gene has a current symbol) and make-gene-info expects a string. 
I think you need to have a condition to check if **curr** is empty list or an opencog atom, and if it is an atom, take its name.

Reply:
The find-genes function from the annotation service is not also working as the responces needs to be casted (it expects integer but now it returns a string '[ ]' or something?

Reply:
@tanksha The latest commit fixes the first issue, now `find-current-symbol` returns gene names. As about the second problem, I will update the annotation-service to handle the changes.